pipelines/folioRancher/folioDevTools/uiManagement/deployUI/Jenkinsfile

```diff
@@ -1 +1,119 @@
-//TBD
+#!groovy
+
+import org.folio.models.Index
+import org.folio.models.InstallRequestParams
+import org.folio.models.RancherNamespace
+import org.folio.rest.GitHubUtility
+import org.folio.models.TenantUi
+import org.folio.rest_v2.Main
+import org.jenkinsci.plugins.workflow.libs.Library
+
+@Library('pipelines-shared-library@RANCHER-741-Jenkins-Enhancements') _
+
+CONFIG_BRANCH = 'RANCHER-741-Jenkins-Enhancements'
+
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '30')),
+    disableConcurrentBuilds(),
+    parameters([
+        folioParameters.cluster(),
+        folioParameters.namespace(),
+        folioParameters.repository(),
+        folioParameters.branch(),
+        folioParameters.configType(),
+        folioParameters.tenantId(),
+        folioParameters.uiBundleBuild(),
+        folioParameters.uiBundleTag(),
+        string(name: 'CUSTOM_HASH', defaultValue: '', description: '(Optional) Commit hash for bundle build from specific commit'),
+        booleanParam(name: 'CONSORTIA', defaultValue: false, description: '(Optional) Include consortia module in UI bundle'),
+        folioParameters.agent(),
+        folioParameters.refreshParameters(),
+    ])
+])
+String defaultTenantId = 'diku'
+
+//Set namespace configuration
+String commitHash = params.CUSTOM_HASH?.trim() ? params.CUSTOM_HASH : common.getLastCommitHash(params.FOLIO_REPOSITORY, params.FOLIO_BRANCH)
+List installJson = new GitHubUtility(this).getEnableList('platform-complete', params.FOLIO_BRANCH)
+TenantUi tenantUi = new TenantUi("${params.CLUSTER}-${params.NAMESPACE}", commitHash, params.FOLIO_BRANCH)
+InstallRequestParams installRequestParams = new InstallRequestParams()
+  .withTenantParameters("loadReference=${params.LOAD_REFERENCE},loadSample=${params.LOAD_SAMPLE}")
+
+RancherNamespace namespace = new RancherNamespace(params.CLUSTER, params.NAMESPACE)
+  .withSuperTenantAdminUser()
+  .withOkapiVersion(params.OKAPI_VERSION)
+  .withDefaultTenant(defaultTenantId)
+  .withDeploymentConfigType(params.CONFIG_TYPE)
+
+namespace.getModules().setInstallJson(installJson)
+namespace.addDeploymentConfig(CONFIG_BRANCH)
+
+namespace.addTenant(folioDefault.tenants()[namespace.getDefaultTenantId()]
+  .withInstallJson(namespace.getModules().getInstallJson().collect())
+  .withIndex(new Index(true, true))
+  .withInstallRequestParams(installRequestParams.clone())
+  .withTenantUi(tenantUi.clone())
+)
+
+Main main = new Main(this, namespace.getDomains()['okapi'], namespace.getSuperTenant())
+List UIModuleList = installJson.findAll { it.id?.startsWith('folio_') == true }
```

Comment:
Let's follow the common naming convention so uiModuleList instead of UIModuleList